lib/system.nim

```diff
@@ -1871,6 +1871,21 @@ proc `$` *[Enum: enum](x: Enum): string {.magic: "EnumToStr", noSideEffect.}
   ## a ``$`` operator for a concrete enumeration is provided, this is
   ## used instead. (In other words: *Overwriting* is possible.)
 
+proc newString*[N](data: array[N, char]): string {.noSideEffect.} =
```

Comment:
I just found out that ``$buffer.cstring`` does already work, so in fact I could simply remove the new ``newString`` proc.